ledger/eval/txntracer.go

```diff
@@ -48,12 +48,28 @@ type StateDeltaSubset struct {
 }
 
 func convertStateDelta(delta ledgercore.StateDelta) StateDeltaSubset {
+	kvmods := make(map[string]ledgercore.KvValueDelta)
```

Comment:
I'm still not sure why we have to copy over the maps here (particularly what recycling the roundCow means)

Maybe an inline comment here will help explain the deep copying too

Reply:
I'll try to add a comment w/ some context. Here's the long version:

The `StateDelta` that we get for each transaction group is stored in a `roundCowState` [here](https://github.com/algorand/go-algorand/blob/master/ledger/eval/cow.go#L75). We create a separate `cow` for each transaction group ([eval.state::child](https://github.com/algorand/go-algorand/blob/master/ledger/eval/eval.go#L958)) and when we finish evaluating the group, the deltas get merged into the parent and the `cow` gets reused for the next group with a call to [cow::recycle](https://github.com/algorand/go-algorand/blob/master/ledger/eval/eval.go#L959). 

The `recycle` function goes through and [resets](https://github.com/algorand/go-algorand/blob/master/ledger/eval/cow.go#L337) the delta which deletes the keys in all of the maps.

Even though the [deltas](https://github.com/algorand/go-algorand/blob/master/ledger/eval/cow.go#L126) function returns by value, the maps are all pointers, so the value returned has pointers to the same objects which will eventually have all of their entries deleted as evaluation progresses.

Reply:
appreciate the detailed context - that's why we are deep copying them right after transaction group has been evaluated but before the round has passed and state has been recycled.